### PR TITLE
Auto delete date fix testing

### PR DESCRIPTION
**Pull request by steffenmartin**
@AndreyNikiforov, here's my first attempt of a comprehensive fix for this bug. I hope this is in adherence to all contributing guidelines.
Since test _test_autodelete_photos_ inside of _tests/test_autodelete_photos.p_y was already performed incorrectly it is somewhat difficult to perform a before-after comparison. If that's needed I suggest to first apply the follow patch to it.
image
This will make it fail in its current state. And once this PR has been merged it will pass again.

### CODE REVIEW

File: tests/test_autodelete_photos.py [hunk 1]
```diff
@@ -128,7 +130,7 @@ def test_download_autodelete_photos(self):
         os.makedirs(base_dir)
 
         files = [
-            "2023/06/06/IMG_3589.JPG"
+            f"{'{:%Y/%m/%d}'.format(datetime.datetime.fromtimestamp(1686106167436.0 / 1000.0, tz=pytz.utc).astimezone(get_localzone()))}/IMG_3589.JPG"
```
Comment (AndreyNikiforov) [inline tests/test_autodelete_photos.py:133]: That is a clever way to to validate logic with localzone. One challenge is still current though -- dependency on runtime environment. E.g. running tests in different timezones (my local machine vs github runners) may produce different results. How can we eliminate it? May be monkeypatch localzone for very specific timezone?

Comment (steffenmartin) [reply tests/test_autodelete_photos.py:133]: I would agree as this seems to be a general issue with all test cases directly or indirectly involving local times (conversions) and not just related to this fix. Best might be to monkeypatch it to the Timezone of the GitHub server in order to minimize risk of existing tests starting to fail. Of course, the better approach in general (and my personal opinion) would be to not do any conversions to local times and using UTC all throughout the code.

Comment (AndreyNikiforov) [reply tests/test_autodelete_photos.py:133]: > I would agree as this seems to be a general issue with all test cases directly or indirectly involving local times (conversions) and not just related to this fix.

I agree that is a general concern, not just this test/fix.

> Best might be to monkeypatch it to the Timezone of the GitHub server in order to minimize risk of existing tests starting to fail

[GitHub runs UTC](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule). Since we are validating that we are taking timezone into account correctly, then picking timezone different from test data will validate a path. May be need to pick a number of timezones to check for edge cases (where date would not change fro the same tests data)?

> Of course, the better approach in general (and my personal opinion) would be to not do any conversions to local times and using UTC all throughout the code.

I am not following your idea. Can you explain a bit more, pls? If asset comes with, say, "2023-01-01 01:23 +0800" timestamp, how would we process it differently from today if we use only UTC in the code?

Side note: This issue with local timezone got me thinking if converting to local timezone instead of using date from the asset is a valid approach at all. E.g. if I celebrated New year in another part of the world, I want these images to be Dec 31 even if my home location at that time was another day. That is a change of the behavior for the product, so we'll only fix the bug for now.
